source/index.html.md

```diff
@@ -1,239 +1,674 @@
+
+### Response
+
+### 200: OK
+
+Query executed successfully.
+
+<button class="see-json">json</button> 
+<button class="see-csv">csv</button> 
+<button class="see-geojson">geojson</button> 
+
+<div class="center-column response-json"></div>
+```
+json 
   {
-    "id": 1,
-    "name": "Fluffums",
-    "breed": "calico",
-    "fluffiness": 6,
-    "cuteness": 7
-  },
+    "results":[
+      {
+        "c0":"foo",
+        "c1":22,
+        "c2":"xx",
+        "c3":"POINT (-79.4 43.7)"
+      },
+      ...
+    ]
+  }
+```
+
+<div class="center-column response-csv"></div>
+```
+csv
+  c0,c1,c2,c3
+  foo,22,xx,POINT (-79.4 43.7)
+  bar,66,yy,POINT (-78.4 42.7)
+  baz,11,xx,POINT (-70.4 35.7)
+```
+
+<div class="center-column response-geojson"></div>
+```
+geojson
   {
-    "id": 2,
-    "name": "Max",
-    "breed": "unknown",
-    "fluffiness": 5,
-    "cuteness": 10
+    "type":"FeatureCollection",
+    "features":[
+      {
+        "type":"Feature",
+        "geometry":{
+          "type":"Point",
+          "coordinates": [ -79.4, 43.7 ]
+        },
+        "properties":{
+          "c0":"foo",
+          "c1":22,
+          "c2":"xx"
+        }
+      },
+      ...
+    ]
   }
-]
 ```
 
-This endpoint retrieves all kittens.
+<!-- placeholder to stop divs from colliding -->
+<div class="response-json"></div>
+<div></div>
+<!-- end placeholder -->
+
+### 422: Unprocessable Entity
+
+Something went wrong with the request.
+
+<div class="center-column"></div>
+```
+{ "error": <Error message> }
+```
+# Query Specification
+
+**NiQL** is modelled after standard SQL, but only supports read-only `SELECT` statements. A traditional query must at minimum have a `SELECT` and `FROM` clause in order to execute. While the `SELECT` clause can be anything, the `FROM` clause must specify a data set and version to query. 
 
-### HTTP Request
+For example, if we were looking to query a data set with an id of `2a6412c0-b3c9-420e-9487-abd21b664ac1` and the version `en-1`, our minimum query would be:
 
-`GET http://example.com/api/kittens`
+<code>SELECT *<br/>
+FROM "2a6412c0-b3c9-420e-9487-abd21b664ac1/en-1" AS BramptonHomes</code>
 
-### Query Parameters
+The parser will verify the table name and handle queries that reference it, but using aliases and quotes whenever possible is recommended.
 
-Parameter | Default | Description
---------- | ------- | -----------
-include_cats | false | If set to true, the result will also include cats.
-available | true | If set to false, the result will include kittens that have already been adopted.
+Below is a summary of supported features:
 
-<aside class="success">
-Remember — a happy kitten is an authenticated kitten!
-</aside>
+## SELECT Features
 
-## Get a Specific Kitten
 
-```ruby
-require 'kittn'
+### COUNT
 
-api = Kittn::APIClient.authorize!('meowmeowmeow')
-api.kittens.get(2)
+<div class="center-column"></div>
+```sql
+{
```

Comment:
Additionally, we could throw these in the curl column and give the example like

```
  {
     "query": "SELECT COUNT(* | column1) FROM data-set-id/en-0"
  }
```

Reply:
Formatting gets a little screwy because of the way the centre column code blocks happen. They don't want to exist on the same line and it looks really ugly. If we wanted a way to show the format and the example, I could go back to code styling using single backticks then show the example on the right column. Whatever you prefer!